### PR TITLE
Features/4 membuat module program studi

### PR DESCRIPTION
**Pull request by juldhais**
ProgramStudi:
- Model
- Controller
- Views (Index/Create/Update/Delete)
- Routing

### CODE REVIEW

File: app/Controllers/ProgramStudi.php [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+namespace App\Controllers;
+use App\Models\ProgramStudiModel;
+
+class ProgramStudi extends BaseController
+{
+    public function index()
+    {
+        if (empty($this->session->get('user_id'))) return redirect("login");
+        return view('users/index');
```
Comment (jacky-htg) [inline app/Controllers/ProgramStudi.php:11]: return view('programstudi/index');

File: app/Controllers/ProgramStudi.php [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+        $sort = $order?$params['order'][0]['dir']:null; // asc or desc
+        $search = $params['search']['value']; // Search value
+        
+        $programStudiModel = new ProgramStudiModel();
+        $count = $programStudiModel->count($search);
+        $data = [
+            "draw" => intval($draw),
+            "iTotalRecords" => $count,
+            "iTotalDisplayRecords" => $count,
+            "data" => $userModel->list($search, $offset, $limit, $order, $sort)
```
Comment (jacky-htg) [inline app/Controllers/ProgramStudi.php:94]: "data" => $programStudiModel->list($search, $offset, $limit, $order, $sort)

File: app/Views/programstudi/create.php [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+                        <select id="akreditasi" class="form-control" name="akreditasi">
+                          <option value="Baik">Baik</option>
+                          <option value="Baik Sekali">Baik Sekali</option>
+                          <option value="Unggul">Unggul</option>
+                        </select>
+                      </div>
+                    </div>
+
+                    <div class="item form-group">
+                      <label class="col-form-label col-md-3 col-sm-3 label-align" for="nama">SK Akreditasi</label>
```
Comment (jacky-htg) [inline app/Views/programstudi/create.php:50]: for="sk_akreditasi"

File: app/Views/programstudi/index.php [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+
+        <script>
+          $(document).ready( function () {
+            let myTable = $('#myTable').DataTable({
+              dom: 'Bfrtip',
+              buttons: [
+                  {
+                      text: '<button>Add New Program Studi</button>',
+                      action: function ( e, dt, node, config ) {
+                          window.location = "<?= base_url('users');?>/create";
```
Comment (jacky-htg) [inline app/Views/programstudi/index.php:54]: window.location = "<?= base_url('programstudi');?>/create";

File: app/Controllers/ProgramStudi.php [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+            }
+        }
+        return view('programstudi/create');
+    }
+
+    public function update($id)
+    {
+        if (empty($this->session->get('user_id'))) return redirect("login");
+        
+        $programStudiModel = new ProgramStudiModel();
```
Comment (jacky-htg) [inline app/Controllers/ProgramStudi.php:42]: tambahkan baris :
```
$data['programstudi'] = $programStudiModel->findById($id);
```

File: app/Views/programstudi/update.php [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+
+            <div class="clearfix"></div>
+
+            <div class="row">
+              <div class="col-md-12 col-sm-12 col-xs-12">
+                <div class="x_panel">
+                  <div class="x_content">
+                  <?= validation_list_errors() ?>
+
+                  <?= form_open("programstudi/{$user['id']}/update"); ?>
```
Comment (jacky-htg) [inline app/Views/programstudi/update.php:19]: <?= form_open("programstudi/{$programstudi['id']}/update"); ?>

**Review by jacky-htg — CHANGES_REQUESTED** [review]
Mas @juldhais tolong difix sesuai komentarku ya... thanks.

File: app/Controllers/ProgramStudi.php (hunk 1 above)
Comment (juldhais) [reply app/Controllers/ProgramStudi.php]: Resolved by the latest commit

File: app/Controllers/ProgramStudi.php (hunk 2 above)
Comment (juldhais) [reply app/Controllers/ProgramStudi.php]: Resolved by the latest commit

File: app/Views/programstudi/create.php (hunk 3 above)
Comment (juldhais) [reply app/Views/programstudi/create.php]: Resolved by the latest commit

File: app/Views/programstudi/index.php (hunk 4 above)
Comment (juldhais) [reply app/Views/programstudi/index.php]: Resolved by the latest commit

File: app/Controllers/ProgramStudi.php (hunk 5 above)
Comment (juldhais) [reply app/Controllers/ProgramStudi.php:42]: Resolved by the latest commit

File: app/Views/programstudi/update.php (hunk 6 above)
Comment (juldhais) [reply app/Views/programstudi/update.php]: Resolved by the latest commit

**Review by jacky-htg — APPROVED** [review]

**Pull request merged**